src/main/sensors/gyro_init.c

```diff
@@ -603,68 +592,69 @@ bool gyroInit(void)
         eepromWriteRequired = true;
     }
 
-#if defined(USE_MULTI_GYRO)
-    if ((gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_BOTH && !((gyroDetectionFlags & GYRO_ALL_MASK) == GYRO_ALL_MASK))
-        || (gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_1 && !(gyroDetectionFlags & GYRO_1_MASK))
-        || (gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_2 && !(gyroDetectionFlags & GYRO_2_MASK))) {
-        if (gyroDetectionFlags & GYRO_1_MASK) {
-            gyro.gyroToUse = GYRO_CONFIG_USE_GYRO_1;
-        } else {
-            gyro.gyroToUse = GYRO_CONFIG_USE_GYRO_2;
+    // check if all enabled sensors are detected
+    for (int i = 0; i < GYRO_COUNT; i++) {
+        if ((gyroDetectionFlags & GYRO_MASK(i)) && (gyroConfig()->gyro_enabled_bitmask & GYRO_MASK(i))) {
+            gyro.gyroEnabledBitmask |= GYRO_MASK(i);
         }
+    }
 
-        gyroConfigMutable()->gyro_to_use = gyro.gyroToUse;
+    if (gyroConfigMutable()->gyro_enabled_bitmask != gyro.gyroEnabledBitmask) {
+        gyroConfigMutable()->gyro_enabled_bitmask = gyro.gyroEnabledBitmask;
         eepromWriteRequired = true;
     }
 
-    // Only allow using both gyros simultaneously if they are the same hardware type.
-    if (((gyroDetectionFlags & GYRO_ALL_MASK) == GYRO_ALL_MASK) && gyro.gyroSensor1.gyroDev.gyroHardware == gyro.gyroSensor2.gyroDev.gyroHardware) {
-        gyroDetectionFlags |= GYRO_IDENTICAL_MASK;
-    } else if (gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_BOTH) {
-        // If the user selected "BOTH" and they are not the same type, then reset to using only the first gyro.
-        gyro.gyroToUse = GYRO_CONFIG_USE_GYRO_1;
-        gyroConfigMutable()->gyro_to_use = gyro.gyroToUse;
+    // Only allow using multiple gyros simultaneously if they are the same hardware type.
+    // Or allow using if they have the same sample rate and scale.
+    bool gyro_hardware_identical = true;
+    uint16_t gyro_sample_rate = 0;
+    float gyro_scale = 0.0f;
```

Comment:
(scale is handled when fusing gyros. Maybe it can be different)

Reply:
Scale is also used to detect if the gyro is overflowing. Hence why at the moment all the scales need to be identical. But you could just pick the first enabled sensors scale to check when checking for overflow (overflow checks for filtered gyro values, not raw, so perhaps that needs to change as well?)

Reply:
Well ... just add comment about this.

Overflow detection should use raw data (it will avoid problems with large offset). 
And with multiple gyros, fusing can help a lot - use weighted gyros, give large priority to low-noise but small range gyro, switch to large-range gyro on fast maneuvers.

(another use os multiple gyros is to use one mechanically filtered gyro for IMU reference - hard-mounted gyro would provide high-bandwidth signal for PIDs etc, soft-mounted gyro will provide beter low-frew imu orientation (mechanical filter will help with aliasing and noise, but introduce considerable delay)